lib/api/redfish-1.0/account-service.js

```diff
@@ -0,0 +1,168 @@
+var Promise = injector.get('Promise'); // jshint ignore:line
+var _ = injector.get('_');  // jshint ignore:line
+var controller = injector.get('Http.Services.Swagger').controller;
+var accountService = injector.get('Http.Services.Api.Account');
+var Errors = injector.get('Errors');
+var assert = injector.get('Assert');
+var configuration = injector.get('Services.Configuration');
+
+var getAccountService = controller(function(req, res) {
+    var options = redfish.makeOptions(req, res);
+    return redfish.render('redfish.1.0.0.accountservice.1.0.0.json', 
+                 'AccountService.1.0.0.json#/definitions/AccountService',
+                  options)
+    .catch(function(error) {
+        return redfish.handleError(error, res);
+    });
+});
+
+var getAccounts = controller(function(req, res) {
+    var options = redfish.makeOptions(req, res);
+    return accountService.listUsers().then(function(users) {
+        options.users = users;
+    })
+    .then(function() {
+        return redfish.render('redfish.1.0.0.manageraccountcollection.json', 
+                 'ManagerAccountCollection.json#/definitions/ManagerAccountCollection',
+                  options);
+    })
+    .catch(function(error) {
+        return redfish.handleError(error, res);
+    });
+});
+
+var getAccount = controller(function(req, res) {
+    var options = redfish.makeOptions(req, res);
+    var name = req.swagger.params.name.value;
+    return accountService.getUserByName(name)
+    .then(function(user) {
+        if(!user) {
+            throw new Error.NotFoundError();
+        }
+        options.user = user;
+        return redfish.render('redfish.1.0.0.manageraccount.1.0.0.json', 
+                 'ManagerAccount.1.0.0.json#/definitions/ManagerAccount',
+                  options);
+    })
+    .catch(function(error) {
+        return redfish.handleError(error, res);
+    });
+});
+
+var createAccount = controller({success: 201}, function(req, res) {
+    var options = redfish.makeOptions(req, res);
+    var payload = req.swagger.params.payload.value;
+    var localUserException = configuration.get('enableLocalHostException', true);
+
+    return Promise.try(function() {
+        assert.string(payload.UserName);
+        assert.string(payload.Password);
+        assert.string(payload.RoleId);
+        return accountService.listUsers();
+    })
+    .then(function(users) {
+        if(!users.length && localUserException && res.locals.ipAddress === '127.0.0.1') {
+            // Only when there are no users, and the remote is a local connection, and we 
+            // permit it, then let them add the first user.
+            return accountService.createUser({ 
+                username: payload.UserName,
+                password: payload.Password,
+                role: payload.RoleId
+            });
+        }
+
+        if( !_.find(users, function(user) {
+            return user.username === payload.UserName;
```

Comment:
You can use the `_.matches` lodash shorthand here `_.find(users, { username: payload.UserName })`:

```
> var l = require('lodash')
> var user = {username: 'test'}
> var payload = {UserName: 'test'}
> l.find([user], {username: payload.UserName})
{ username: 'test' }
> l.find([user], {username: 'notfound'})
undefined
```